internal/gensupport/send.go

```diff
@@ -48,8 +48,24 @@ func SendRequest(ctx context.Context, client *http.Client, req *http.Request) (*
 	if ctx != nil {
 		headers := callctx.HeadersFromContext(ctx)
 		for k, vals := range headers {
-			for _, v := range vals {
-				req.Header.Add(k, v)
+			if k != "x-goog-api-client" {
+				for _, v := range vals {
```

Comment:
nit: could you flip this conditional? Seems more readable to have the default case as the "else".